src/ethereum/EnigmaContractProductionWriterAPI.js

```diff
@@ -462,23 +464,22 @@ class EnigmaContractProductionWriterAPI extends EnigmaContractWriterAPI {
   }
 
   async _parsePastEvents(eventName, filter, blockNumber) {
-    const rawEvents = await this._enigmaContract.getPastEvents(eventName, {
-      fromBlock: blockNumber,
-      filter: filter
-    });
-    let events = {};
-    if (Array.isArray(rawEvents) && rawEvents.length > 0) {
-      if (rawEvents.length > 1) {
-        this._logger.info(
-          `Received am unexpected number of events for ${eventName} with the current filter ${JSON.stringify(
-            filter
-          )}.. taking the first`
-        );
-      }
-      events[eventName] = rawEvents[0];
-      events = this._parseEvents(events);
+    const rawEvents = await this._enigmaContract.getPastEvents(eventName, { fromBlock: blockNumber, filter: filter });
+
+    if (!Array.isArray(rawEvents) || rawEvents.length === 0) {
+      return {};
     }
-    return events;
+
+    if (rawEvents.length > 1) {
```

Comment:
Perhaps with this change, we should more strict in the case where more than event is received (is there a real case in which this can happen?) and reject it? What do you think?

Reply:
This depends on the invoked contract function.
I think in this case `deploySecretContract` emits only one event.